src/plugins/discover/server/ui_settings.ts

```diff
@@ -171,6 +173,38 @@ export const getUiSettings: () => Record<string, UiSettingsParams> = () => ({
       name: 'discover:useLegacyDataGrid',
     },
   },
+  [AGGREGATED_VIEW_SETTING]: {
+    name: i18n.translate('discover.advancedSettings.aggregatedViewName', {
+      defaultMessage: 'Aggregated view',
+    }),
+    value: false,
+    description: i18n.translate('discover.advancedSettings.aggregatedViewDescription', {
+      defaultMessage:
+        'Turn on this option to show the aggregated view by default. Turn off to show the document view.',
+    }),
+    category: ['discover'],
+    schema: schema.boolean(),
+    metric: {
+      type: METRIC_TYPE.CLICK,
+      name: 'discover:aggregatedView',
+    },
+  },
+  [AGGREGATED_VIEW_PREVIEW]: {
+    name: i18n.translate('discover.advancedSettings.showAggregatedPreviewName', {
+      defaultMessage: 'Show aggregated preview',
+    }),
+    value: false,
+    description: i18n.translate('discover.advancedSettings.showAggregatedPreviewDescription', {
+      defaultMessage: 'Turn on this option to show the preview in the aggregated view by default.',
+    }),
+    category: ['discover'],
+    schema: schema.boolean(),
+    metric: {
+      type: METRIC_TYPE.CLICK,
+      name: 'dataVisualizerTable:showPreview',
+    },
+  },
```

Comment:
We tend in the past to overconfigure our UIs with advanced settings that eventually caused problem with having a harder time removing them, overly complex (untested) default state configuration, etc. I am not sure we really want to give advanced settings to let the user configure the default state here.

Could we please explain a bit on the user-story and use-cases here why we think this needs to be configurable by the user. cc @andreadelrio @ryankeairns @shaunmcgough

Reply:
If we stored the view mode in the URL as suggested in https://github.com/elastic/kibana/pull/107184#issuecomment-917032541, then there is probably no need for the `Aggregated view` setting. 

The only reason I can see for having the `Show aggregated preview` setting is if we're concerned that for very large indices that the extra queries used to populate the histograms would place too much load on the cluster. Maybe with the follow up work you have planned @qn895 to implement search strategies will remove the need for this advanced setting?

Reply:
I have moved both settings to Discover app state in https://github.com/elastic/kibana/pull/107184/commits/bed2cd6a78eabfb1626ac599dbbe6d62e6fe77d1 and https://github.com/elastic/kibana/pull/107184/commits/f29765bfb5902237c2f5f50b5d7a22bae970b12c This change should resolve Shaun's issue of having having the view reseted after navigating to Lens, and should also persist the configurations when the search is saved.